### PR TITLE
Signed release packages using Github artifact attestation

### PR DESCRIPTION
**Pull request by egecetin**
Recently GitHub release public beta for artifact signing https://github.blog/2024-05-02-introducing-artifact-attestations-now-in-public-beta/. With this PR, generated artifacts should also generate `.sigstore.json` files to verify packages. It should be ready to review. Since I changed permissions from a PR, looks like write permissions prohibited. I can't find a way to tweak it to get passing result from CI.

Note: Unfortunately requires drop of Ubuntu 18.04 support because of node version

- All attestations can be found in summaries: https://github.com/egecetin/PcapPlusPlus/actions/runs/9702477673 
- Example attestation page: https://github.com/egecetin/PcapPlusPlus/attestations/1193027
- Example release page: https://github.com/egecetin/PcapPlusPlus/releases/tag/v24.1

### CODE REVIEW

**Comment by egecetin** [comment]
@seladb I updated this PR and finished work. Only need (or dependency) is #1408. After the merge of #1408 just need to test with Ubuntu 24.04 and it is ready to merge. You can review this PR when you have time.

**Comment by seladb** [comment]
@egecetin we removed Ubuntu 18.04, do you think you can update the PR and resolve conflicts?
If the PR is ready for review I can review it

**Comment by egecetin** [comment]
@seladb It is ready

**Comment by tigercosmos** [comment]
@egecetin  seems the CI have some errors?

**Comment by egecetin** [comment]
@tigercosmos It is because of permissions. Since the write permissions set from a PR it prohibits the write permission so, it fails. You can check my master branch or run from my test tag  https://github.com/egecetin/PcapPlusPlus/actions/runs/9858969300

**Comment by tigercosmos** [comment]
> @tigercosmos It is because of permissions. Since the write permissions set from a PR it prohibits the write permission so, it fails. You can check my master branch or run from my test tag https://github.com/egecetin/PcapPlusPlus/actions/runs/9858969300

I see. it's cool!

**Review by tigercosmos — APPROVED** [review]

**Comment by egecetin** [comment]
@seladb This PR is also ready for review but what do you think about it? Merge before release or after release? It is nice to merge before release but I'm ok for both options since already have many changes for release and last release was long time ago...

**Review by clementperon — APPROVED** [review]

File: .github/workflows/package.yml [hunk 1]
```diff
@@ -139,6 +157,12 @@ jobs:
       - name: Package
         run: cmake --build "$BUILD_DIR" --target package
 
+      - name: Generate artifact attestation
+        id: attestation
```
Comment (clementperon) [inline .github/workflows/package.yml:161]: For my culture why this job need an id ?

Comment (egecetin) [reply .github/workflows/package.yml:161]: To be honest, I don't know also. Their example has this structure so i just copied 😁

Comment (clementperon) [reply .github/workflows/package.yml:161]: I think you can drop it if you don't use a variable like ${{ steps.attestation.XXXX }}

Comment (seladb) [reply .github/workflows/package.yml:161]: yes I agree 👍

**Comment by seladb** [comment]
> @seladb This PR is also ready for review but what do you think about it? Merge before release or after release? It is nice to merge before release but I'm ok for both options since already have many changes for release and last release was long time ago...

@egecetin I'm ok with merging this PR before the next release

**Review by seladb — APPROVED** [review]
Fix this comment: https://github.com/seladb/PcapPlusPlus/pull/1390#discussion_r1729467931 and merge!

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # We need 3.12 or later, so that we can set policy CMP0074; see below.
 cmake_minimum_required(VERSION 3.12)
 
-set(PCAPPP_VERSION "23.09+")
```
[the remark this answers is not in the document]
Comment (egecetin) [reply CMakeLists.txt:4]: Oops this for testing from my tags. I should revert it

[the remark this answers is not in the document]
Comment (egecetin) [reply CMakeLists.txt:4]: I generated a test tag again. If everything passes I'll revert it

**Comment by egecetin** [comment]
@seladb last test run: https://github.com/egecetin/PcapPlusPlus/actions/runs/10544926628

File: CMakeLists.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (egecetin) [reply CMakeLists.txt]: Fixed with e5e08cd

**Comment by seladb** [comment]
> @seladb last test run: https://github.com/egecetin/PcapPlusPlus/actions/runs/10544926628

Where are the packages created? This tag doesn't have any packages attached... 🤔

**Comment by egecetin** [comment]
It was draft. Released now

**Comment by seladb** [comment]
> It was draft. Released now
> 
> https://github.com/egecetin/PcapPlusPlus/releases/tag/v24.2

@egecetin I couldn't find the `.sigstore.json` file in the atrifacts 🤔

**Comment by egecetin** [comment]
@seladb I removed them because GitHub stores them in attestation tab already (Check [here](https://github.com/egecetin/PcapPlusPlus/attestations)) For verifying there is no need to download `.sigstore.json`. In old commits I added them but then noticed a future problem. GitHub generates them as `jsonl`. This means that every attestation stored in lines of a file. If we add deb/rpm packaging support there will be more than one packages per platform. So we should distinguish sigstore files from lines and to do this we should add a script to parse which line corresponds to which package.

File: README.md [hunk 3]
```diff
@@ -96,6 +96,20 @@ git clone https://github.com/seladb/PcapPlusPlus.git
...
+
+```shell
+gh attestation verify <path-to-package-file> --repository seladb/PcapPlusPlus
+```
+
+and you should see the following output in your terminal:
+
+```shell
+✓ Verification succeeded!
+```
```
Comment (seladb) [inline README.md:111]: 1. I think it'd be helpful to add a link to the attestation page : https://github.com/seladb/PcapPlusPlus/attestations
2. We should also add it to PcapPlusPlus web-site, maybe here? https://pcapplusplus.github.io/docs/next/install#github-release

**Review by seladb — APPROVED** [review]
Please see 1 comment, otherwise LGTM

**Pull request merged**